### PR TITLE
Add Kyber Network token v2

### PR DESCRIPTION
**Pull request by manhlx3006**

### CODE REVIEW

File: contracts/sol6/token/KyberNetworkTokenV2.sol [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        super._mint(account, amount);
+        emit Minted(account, amount, minter);
+    }
+
+    /// @dev burn old knc and mint new knc for msg.sender, ratio 1:1
+    function mintWithOldKnc(uint256 amount) external {
+        assert(IERC20Burnable(oldKNC).burnFrom(msg.sender, amount));
+
+        super._mint(msg.sender, amount);
+        emit BurntAndMinted(msg.sender, amount);
```
Comment (ducquangkstn) [inline contracts/sol6/token/KyberNetworkTokenV2.sol]: I prefer to use "migrate"

File: contracts/sol6/token/KyberNetworkTokenV2.sol [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    function mint(address account, uint256 amount) external onlyMinter {
+        super._mint(account, amount);
+        emit Minted(account, amount, minter);
+    }
+
+    /// @dev burn old knc and mint new knc for msg.sender, ratio 1:1
+    function mintWithOldKnc(uint256 amount) external {
+        assert(IERC20Burnable(oldKNC).burnFrom(msg.sender, amount));
```
Comment (Anyhowclick) [inline contracts/sol6/token/KyberNetworkTokenV2.sol]: oldKNC has 1 assert already, i feel it isn't really necessary to add another. Help save gas..? =)

File: contracts/sol6/token/KyberNetworkTokenV2.sol [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20Burnable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+interface IERC20Burnable {
+    function burnFrom(address _from, uint256 _value) external returns (bool);
+}
+
+contract KyberNetworkTokenV2 is ERC20Burnable, Ownable {
```
Comment (ducquangkstn) [inline contracts/sol6/token/KyberNetworkTokenV2.sol]: Not in this PR, but do we want ERC20Permit

**Review by Anyhowclick — COMMENTED** [review]
Small thing, but I suggest we standardise the casing of test titles? Whether the first letter should be uppercase.

Eg. `Test mint reverts not minter` VS `test mint reverts not minter`

File: contracts/sol6/mock/Token.sol [hunk 4]
```diff
@@ -220,8 +220,9 @@ contract Token is StandardToken {
     }
 
     // save some gas by making only one contract call
-    function burnFrom(address _from, uint256 _value) public returns (bool) {
-        transferFrom(_from, msg.sender, _value);
+    function burnFrom(address _from, uint256 _value) public
+        returns (bool) {
+        assert( transferFrom( _from, msg.sender, _value ) );
```
Comment (Anyhowclick) [inline contracts/sol6/mock/Token.sol:225]: Remove the spaces in the assert statement, ie. `assert(transferFrom( _from, msg.sender, _value ));`

File: contracts/sol6/token/KyberNetworkTokenV2.sol (hunk 2 above)
Comment (manhlx3006) [reply contracts/sol6/token/KyberNetworkTokenV2.sol]: yeah it's true, actually it is either assertion or return true for burnFrom in old KNC contract haha

**Review by Anyhowclick — DISMISSED** [review]

File: test/sol6/kyberNetworkTokenV2.js [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+                let amount = new BN(10).pow(new BN(18));
+                await newKNC.mint(user, amount, { from: minter });
+                let userBalance = await newKNC.balanceOf(user);
+                await expectRevert(
+                    newKNC.burn(userBalance.add(new BN(1)), { from: user }),
+                    "ERC20: burn amount exceeds balance"
+                );
+            });
+
+            it(`test data changes`, async() => {
```
Comment (ducquangkstn) [inline test/sol6/kyberNetworkTokenV2.js]: Pls rewrite this

Comment (ducquangkstn) [reply test/sol6/kyberNetworkTokenV2.js]: Also add test totalSupply

**Comment by manhlx3006** [comment]
Updated:
- Remove redundant tests for approve
- Update tests for burn
- Event BurntAndMinted -> Migrated
- totalSupply, balance, allowance should be tested in most of tests to check correct data changes, so we don't need tests for it.

**Review by Anyhowclick — DISMISSED** [review]

**Review by ducquangkstn — APPROVED** [review]

**Pull request merged**